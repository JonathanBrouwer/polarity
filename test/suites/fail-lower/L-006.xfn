data Bool { True, False }

codata Pair(a b: Type) {
    Pair(a,b).fst(a b: Type) : a,
    Pair(a,b).snd(a b: Type) : b
}

data Unit { MkUnit }

def Unit.foo : Pair(Bool,Bool) {
    MkUnit => comatch { fst(a,b) => True, snd(a,b) => True}
}