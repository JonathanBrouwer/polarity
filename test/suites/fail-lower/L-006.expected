Local (co)match must be named